backends/bmv2/common/control.cpp

```diff
@@ -93,10 +93,10 @@ void ControlConverter::convertTableEntries(const IR::P4Table *table,
                     key->emplace("start", stringRepr(0, k8));
                     key->emplace("end", stringRepr((1 << keyWidth)-1, k8));  // 2^N -1
                 } else {
-                    ::error("%1% invalid range key expression", k);
+                    ::error(ErrorType::ERR_UNSUPPORTED, k, "range key expression");
                 }
             } else {
-                ::error("unkown key match type '%1%' for key %2%", matchType, k);
+                ::error(ErrorType::ERR_UNKNOWN, k, "key match type " + matchType);
```

Comment:
Whenever you converted something to a concatenation you will get a different result; in particular you won't information about the source line position of the construct which is appended. Sometimes this may be desired.

Reply:
The line number is always printed for the first argument (k in this case).